sdk/appconfiguration/app-configuration/samples/react-sample/src/components/Page.tsx

```diff
@@ -0,0 +1,200 @@
+/* eslint-disable jsx-a11y/anchor-is-valid */
+/*
+  Copyright (c) Microsoft Corporation.
+  Licensed under the MIT license.
+*/
+
+import React from "react";
+import { useEffect, useState } from "react";
+import { getEnvironmentVariable } from "../utils";
+import {
+  AppConfigurationClient,
+  featureFlagPrefix,
+  isFeatureFlag,
+  isFeatureFlagClientFilter
+} from "@azure/app-configuration";
+import { InteractiveBrowserCredential } from "@azure/identity";
+
+let client: AppConfigurationClient;
+
+const featureFlag1 = {
+  id: "react-app-feature-1",
+  description: "",
+  enabled: false,
+  conditions: {
+    client_filters: []
+  }
+};
+const featureFlag2 = {
+  id: "react-app-feature-2",
+  description: "",
+  enabled: true,
+  conditions: {
+    client_filters: [
+      {
+        name: "Microsoft.TimeWindow",
+        parameters: {
+          Start: "Mon, 12 Apr 2021 07:00:00 GMT",
+          End: "Wed, 14 Apr 2021 07:00:00 GMT"
+        }
+      }
+    ]
+  }
+};
+
+export default function Page(): JSX.Element {
+  const feature1Name = "react-app-feature-1";
+  const feature2Name = "react-app-feature-2";
+  const [feature1, setFeature1] = useState<{ enabled: boolean }>({ enabled: false });
+  const [feature2, setFeature2] = useState<{ enabled: boolean }>({ enabled: false });
+
+  const getFeatureFlags = async (keys: string[]): Promise<void> => {
+    const [setting1, setting2] = await Promise.all(
+      keys.map((key: string) =>
+        client.getConfigurationSetting({
+          key: featureFlagPrefix + key
+        })
+      )
+    );
+    if (isFeatureFlag(setting1)) {
+      // console.log(`${setting1.key} is enabled : ${setting1.enabled}`, setting1);
+      setFeature1({ enabled: setting1.enabled });
+    }
+    if (isFeatureFlag(setting2)) {
+      // console.log(`${setting2.key} is enabled : ${setting2.enabled}`, setting2);
+      const clientFilter = setting2.conditions.clientFilters?.[0];
+      if (isFeatureFlagClientFilter("timeWindow", clientFilter)) {
+        const now = Date.now();
+        const withinRange =
+          now - Date.parse(clientFilter.parameters.start) > 0 &&
+          Date.parse(clientFilter.parameters.end) - now > 0;
+        setFeature2({ enabled: withinRange });
+      }
+    }
+  };
+
+  useEffect(() => {
+    const endpoint = getEnvironmentVariable("REACT_APP_APP_CONFIG_ENDPOINT");
+    const clientId = getEnvironmentVariable("REACT_APP_AZURE_CLIENT_ID");
+    const tenantId = getEnvironmentVariable("REACT_APP_AZURE_TENANT_ID");
+    const credential = new InteractiveBrowserCredential({ clientId, tenantId });
```

Comment:
I see this is using IBC, in that case don't we need instructions on setting up the app registration as well configuring the right permissions to access these app configuration?

Reply:
They are in the root readme, I can point them to it.

Reply:
Added more text w.r.t API permissions and the RBAC/